### PR TITLE
SPEC-1666 Clarify the conditions for adding the retryable writes error label during transactions

### PR DESCRIPTION
**Pull request by egiurleo**
Clarify that the RetryableWriteError label should not be added to errors that occur during writes within a transaction. @divjotarora did a POC and confirmed that these changes work in Go.

### CODE REVIEW

File: source/transactions/tests/error-labels.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ tests:
         result:
           # Note, the server will return the errorLabel in this case.
           errorLabelsContain: ["TransientTransactionError"]
-          errorLabelsOmit: ["UnknownTransactionCommitResult"]
+          errorLabelsOmit: ["RetryableWriteError", "UnknownTransactionCommitResult"]
```
Comment (egiurleo) [inline source/transactions/tests/error-labels.yml:89]: This test should already pass, just including the RetryableWriteError label to make it consistent with the rest of the tests in this file.

File: source/transactions/tests/error-labels.yml [hunk 2]
```diff
@@ -143,7 +143,7 @@ tests:
...
```
Comment (egiurleo) [inline source/transactions/tests/error-labels.yml:146]: This test should already pass, just including the RetryableWriteError label to make it consistent with the rest of the tests in this file.

File: source/transactions/tests/error-labels.yml [hunk 3]
```diff
@@ -200,7 +200,7 @@ tests:
...
```
Comment (egiurleo) [inline source/transactions/tests/error-labels.yml:203]: This test should already pass, just including the RetryableWriteError label to make it consistent with the rest of the tests in this file.

**Comment by divjotarora** [comment]
Discussed offline and @egiurleo is going to see if we can add tests to cause write concern errors during transactions. In the Go POC, the current tests passed even if I only update the code to check txn requirements in the command error case, not the write concern error case.

**Comment by egiurleo** [comment]
Will work on this shortly and ping both reviewers.

**Comment by DmitryLukyanov** [comment]
I can confirm that the currently provided tests pass on c# driver

File: source/transactions/tests/error-labels.json [hunk 4]
```diff
@@ -1098,6 +1105,106 @@
...
+          "object": "collection",
+          "arguments": {
+            "session": "session0",
+            "document": {
+              "_id": 1
+            }
+          },
+          "result": {
+            "errorLabelsContain": [
+              "TransientTransactionError"
```
Comment (DmitryLukyanov) [inline source/transactions/tests/error-labels.json:1148]: The error 91 doesn't look like a network error. Should this label really be added?

Comment (egiurleo) [reply source/transactions/tests/error-labels.json:1148]: You're right, this is a typo! Thanks for catching. Fixing this now.

File: source/transactions/tests/error-labels.yml [hunk 5]
```diff
@@ -670,6 +673,68 @@ tests:
...
+          command:
+            abortTransaction: 1
+            lsid: session0
+            txnNumber:
+              $numberLong: "1"
+            startTransaction:
+            autocommit: false
+            writeConcern:
+          command_name: abortTransaction
+          database_name: *database_name
```
Comment (divjotarora) [inline source/transactions/tests/error-labels.yml:732]: This expects the command to go to the `transactions-tests` database but per https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#aborttransaction-1, `abortTransaction` must be sent to `admin`.

File: source/transactions/tests/error-labels.yml [hunk 6]
```diff
@@ -670,6 +673,68 @@ tests:
...
+          writeConcernError:
+            code: 91
+            errmsg: Replication is being shut down
+
+    operations:
+      - name: startTransaction
+        object: session0
+        arguments:
+          options:
+            writeConcern:
```
Comment (divjotarora) [inline source/transactions/tests/error-labels.yml:692]: This configures the transaction to use write concern majority, but the `abortTransaction` expectation below expects `writeConcern: `. Per https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#writeconcern, I think the expectation should include majority. Alternatively, we could just remove the `writeconcern` field from the expectation. That seems like something that should be tested by the options tests, not this.

Comment (egiurleo) [reply source/transactions/tests/error-labels.yml:692]: Yup, agreed.

File: source/transactions/tests/error-labels.yml (hunk 5 above)
Comment (egiurleo) [reply source/transactions/tests/error-labels.yml:732]: 👍

**Pull request merged**